solr/core/src/java/org/apache/solr/cloud/api/collections/CreateCollectionCmd.java

```diff
@@ -321,6 +319,8 @@ public void call(ClusterState clusterState, ZkNodeProps message, @SuppressWarnin
         }
       }
       if(isPRS) {
+        byte[] data = Utils.toJSON(Collections.singletonMap(collectionName, clusterState.getCollection(collectionName)));
+        ocmh.zkStateReader.getZkClient().setData(collectionPath, data, true);
```

Comment:
Is there a good reason that we don't update state.json using `ocmh.overseer.offerStateUpdate` (which is single threaded) ? Doing directly writes here opens up possibility of race condition - even if it does not happen right now, it's a liability. 

`ocmh.overseer.offerStateUpdate` enforces ordering, load throttling/tracking and most importantly single threading to eliminate any race conditions on `state.json` updates. (uberseer could still contend for state.json update, but that's outside of this scope)

Reply:
> Is there a good reason that we don't update state.json using `ocmh.overseer.offerStateUpdate`

Yes. `ocmh.overseer.offerStateUpdate` is a command and we should be using it only for well known commands. Here we are trying to update the whole state in one go